### ISSUE TITLE
Rich text template editor

### ISSUE THREAD

**Issue by andrewleith** [cds-snc/notification-planning#1496]
## Description
As a user, I need a rich text editor to be able to easily create and edit templates so that I can get to sending notifications faster.

### WHY are we building?
- Create a simpler template-building experience.
### WHAT are we building?
- A WYSIWYG template editor
### VALUE created by our solution
- Easier to see in one toolbar what features our templates offer
- Visualize how notifications will look without leaving the editor

## Documentation and Artifacts
![image](https://github.com/cds-snc/notification-planning/assets/823749/2a2379a3-92fc-419d-99c4-9177ac797927)
https://miro.com/app/board/uXjVMgJrBHc=/
https://app.zenhub.com/workspaces/notify-planning-614b3ad91bc2030015ed22f5/issues/gh/cds-snc/notification-planning/1496
[Notes from Fable QA Sessions](https://docs.google.com/document/d/16DCWoenNnrLAnLZJby5I6xFqfTOTsY-hazWjECQSvdE/edit)
Exploring Front End options Quill and Lexical: https://app.zenhub.com/workspaces/notify-planning-614b3ad91bc2030015ed22f5/issues/gh/cds-snc/notification-planning/2083

## Acceptance Criteria
Given a GC Notify client isn't familiar with markdown and wants to format their template quickly and easily, when they build their template in the GC Notify UI, then they have a simple rich text editor they can use that has parity with our current markdown editing options.  

- [X] ~does not~ include keyboard shortcuts (update- we ended up including these anyway)
- [X] does not introduce new rich text editing options, only parity with current formatting options
- [X] keeps toggle for right-to-left text formatting and incorporates it into the new editor
- [X] An option to add the bilingual message formatting structure with the selection of one button for time-saving and ease of use

### Dev
- [ ] ~Can save to our DB as a template. Like a template v2, where v1 is our current template format~ (Unsure, probably?)
- [ ] ~Can interpolate placeholder during the sending flow, with spreadsheet or as a single notification~ (Unsure, probably?)
- [X] Doesn't conflict with our render_template method when previewing or sending an email
- [X] Generates email worthy HTML markup with inlined styles. 

### Design
- [X] 1. Alternative navigation: if using voice commands - the buttons don't work
       (might do an easy fix for V1 - proper label text instead of aria, and then come back to it v2 potential testing)
- [X] 2. Design for new ~Lexical~ Tip Tap Interface @amazingphilippe to design
       Button to opt-in to new editor from current editor
       Button to go back to the old editor from the new editor
- [X] 3 .Accessibility: Inscreen dialogue for link editor - @amazingphilippe to make a design
- [X] Design to Set the default to the old editor and a toggle to move.

### Growth/ PM
- [ ] ~Do we need to add Guidance for the new RTE? (there is a help section in the toolbar and it can be used - but doesn't have to be)~ Answer: decided we don't need this now.
- [X] Coordinate re New Feature Plan (coordinate with Growth) (Growth is aware; will be planning to promote this spring)

### Measurements and insights Plan:
- [X] Plan for Google Analytics
- [X] Plan for Tickets in freshdesk
- [X] Compare how many people use the old editor vs the new editor (we have initial metrics; 107 made use of RTE right away; will continue to measure in future quarters)
- [ ] ~Compare how many people save a template with the old editor vs the new editor~

## QA Steps
- [X] Tested in a realistic production scenario

## Research findings
[Some participants expected to format text by using a toolbar or by selecting a word and seeing a styling menu (e.g. Word) For these participants, it wasn't good to learn something in order to complete the task. For them, where to find help wasn't obvious, but they successfully clicked the guide links because they didn't see many other options.](https://airtable.com/appWwAmHwDLtpIyko/tbl38n7ofWYBuezFc/viwEUFgBd0KLqH7oo/recFD9ZeORJ2aWiIO?blocks=hide)

[Participants noted that Markdown isn't something they/their colleagues have knowledge about. However if they themselves had technical knowledge, they were happy to find markdown as the formatting language.](https://airtable.com/appWwAmHwDLtpIyko/tbl38n7ofWYBuezFc/viwEUFgBd0KLqH7oo/recYHJitUFEoplqXJ?blocks=hide)

**Comment by jzbahrai** [comment]
Andrew to demo both editors and then one other dev and andrew to work on this together to move the project foward.

**Comment by yaelberger-commits** [comment]
Touch base with dave S on Forms for insights implementing their rich text editor

**Comment by schancase** [comment]
Closing as 100% done with Carolyn. We did a detailed review of all the A/C checkboxes above and provided updates.
